src/fitnesse/junit/JUnitXMLTestResultRecorder.java

```diff
@@ -0,0 +1,118 @@
+package fitnesse.junit;
+
+import org.apache.commons.text.StringEscapeUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Recorder for test results in the JUnit xml format.
+ * The xml report is written to the file system.
+ */
+public class JUnitXMLTestResultRecorder {
+
+  private final File reportsDir;
+
+  /**
+   * Constructs a JUnitXMLTestResultRecorder.
+   * @param reportsDir directory the test results are to be written to (eg. <code>reports.junitXml.destination</code>)
+   */
+  public JUnitXMLTestResultRecorder(File reportsDir) {
+    this.reportsDir = reportsDir;
+    this.reportsDir.mkdirs();
+  }
+
+  /**
+   * Records result of single page (i.e. test).
+   * @param testName name of test
+   * @param skipped number of skipped tests
+   * @param failures number of test failures
+   * @param errors number of errors of test
+   * @param throwable throwable from test
+   * @param executionTime execution time in seconds
+   */
+  void recordTestResult(String testName, int skipped, int failures, int errors, Throwable throwable, double executionTime) throws
```

Comment:
I think passing in the executionTime as long will be better (you can divide by 1000 here in this code), FitNesse's `TimeMeasurement.elapsed()` should provide in that format also

Reply:
Fixed. Please revise.